docs/ovs-pipeline.md

```diff
@@ -0,0 +1,561 @@
+  more information.
+* *conntrack*: a connection tracking module that can be used by OVS to match on
+  the state of a TCP, UDP, ICMP, etc., connection. See the [OVS Conntrack
+  tutorial](http://docs.openvswitch.org/en/latest/tutorials/ovs-conntrack/) for
+  more information.
+* *dmac table*: a traditional L2 switch has a "dmac" table which maps
+  learned destination MAC address to the appropriate egress port. It is often
+  the same physical table as the "smac" table (which matches on the source MAC
+  address and initiate MAC learning if the address is unknown).
+
+## Registers
+
+We use 2 32-bit OVS registers to carry information throughout the pipeline:
+ * reg0 (NXM_NX_REG0):
+   - bits [0..15] are used to store the traffic source (from tunnel: 0, from
+   local gateway: 1, from local Pod: 2). It is set in the [ClassifierTable].
+   - bit 16 is used to indicate whether the destination MAC address of a packet
+   is "known", i.e. corresponds to an entry in [L2ForwardingCalcTable], which is
+   essentially a "dmac" table.
+ * reg1 (NXM_NX_REG1): it is used to store the egress OF port for the packet and
+ is set by [L2ForwardingCalcTable].
+
+## Network Policy implementation
+
+Several tables of the pipeline are dedicated to [K8s Network
+Policy](https://kubernetes.io/docs/concepts/services-networking/network-policies/)
+implementation ([EgressRuleTable], [EgressDefaultTable], [IngressRuleTable] and
+[IngressDefaultTable]).
+
+The Antrea implementation of K8s Network Policy, including the communication
+channel between the Controller and Agents, and how a Network Policy is mapped to
+OVS flows at each Node, will be described in details in a separate document. For
+the present document, we will use the Network Policy example below, and explain
+how these simple ingress and egress rules map to individual flows as we describe
+the relevant tables of our pipeline.
+
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: test-network-policy
+  namespace: default
+spec:
+  podSelector:
+    matchLabels:
+      app: nginx
+  policyTypes:
+  - Ingress
+  - Egress
+  ingress:
+  - from:
+    - podSelector:
+        matchLabels:
+          app: nginx
+    ports:
+    - protocol: TCP
+      port: 80
+  egress:
+  - to:
+    - podSelector:
+        matchLabels:
+          app: nginx
+    ports:
+    - protocol: TCP
+      port: 80
+```
+
+This Network Policy is applied to all Pods with the `nginx` app label in the
+`default` namespace. For these Pods, it only allows TCP traffic on port 80 from
+and to Pods which also have the `nginx` app label. Because Antrea will only
+install OVS flows for this Network Policy on Nodes for which some of the Pods
+are the target of the policy, we have scheduled 2 `nginx` Pods on the same
+Node. They received IP addresses 10.10.1.2 and 10.10.1.3 from the Antrea CNI, so
+you will see these addresses show up in the OVS flows.
+
+## Tables
+
+![OVS pipeline](/docs/assets/ovs-pipeline.svg)
```

Comment:
in the diagram i noticed an arrow from table70 -> table60.. i believe it should be the other way around